src/extension.ts

```diff
@@ -127,9 +126,13 @@ export function activate(context: vscode.ExtensionContext) {
             // prevent further popups.
             notifiedFiles.add(notification.origin());
           }
+          if (notification.isDone()) {
+          record(TelemetryActions.componentAnalysisTriggered, {depFileName: path.basename(vscode.window.activeTextEditor.document.fileName)});
+          }
         });
 
         lspClient.onNotification('caError', respData => {
+          record(TelemetryActions.componentAnalysisFailed, {error: respData.error});
           const notification = new CANotification(respData);

```

Comment:
Add filename payload for error also.
```suggestion
          const notification = new CANotification(respData);
          record(TelemetryActions.componentAnalysisFailed, {filename: path.basename(notification.origin()), error: respData.data()});
```